frame/evm/src/runner/stack.rs

```diff
@@ -141,13 +144,28 @@ where
 		// of a precompile. While mainnet Ethereum currently only has stateless precompiles,
 		// projects using Frontier can have stateful precompiles that can manage funds or
 		// which calls other contracts that expects this precompile address to be trustworthy.
-		if is_transactional
-			&& (!<AccountCodes<T>>::get(source).is_empty() || precompiles.is_precompile(source))
-		{
-			return Err(RunnerError {
-				error: Error::<T>::TransactionMustComeFromEOA,
-				weight,
-			});
+		if is_transactional {
+			let is_precompile = match precompiles.is_precompile(source, gas_limit)
```

Comment:
You can simplify a lot this part because we decided to not check precompiles here, @nanocryk can confirm

Reply:
Since we already decided to not check precompiles here, and your PR add a lot of complexity due to that, I'm more inf avor of removing this complexity right now